src/en/charms-storage.md

```diff
@@ -226,22 +246,27 @@ juju deploy postgresql --attach-storage pgdata/0
 !!! Note:
     The `--attach-storage` and `-n` flags cannot be used together.
 
-Add a new unit of the postgresql application, attaching existing storage
-'pgdata/0' to the new unit:
+To add a new Ceph OSD unit with (detached) existing storage 'osd-devices/2':
 
 ```bash
-juju add-unit postgresql --attach-storage pgdata/0
+juju add-unit ceph-osd --attach-storage osd-devices/2
 ```
 
-Destroy already detached storage 'pgdate/0' (remove it from the model):
+To destroy already detached storage 'osd-devices/3' (remove it from the model):
 
 ```bash
-juju remove-storage pgdata/0
+juju remove-storage osd-devices/3
 ```
 
-If an attempt is made to remove storage that is currently in use (i.e. it is
-attached) Juju will return an error.
+To upgrade the OSD journal of Ceph unit 'ceph-osd/0' from magnetic to solid
+state (SSD) and remove the uneeded original journal 'osd-journals/0':
 
+```bash
+juju add-storage ceph-osd/0 osd-journals=ebs-ssd,8G,1
```

Comment:
Your question tells me the page isn't clear (my failure). Yes, the charm is the limiting factor to everything you can do to the application, including storage. To me, this is answered in the introductory section:

> Certain applications can benefit from advanced storage configurations and if a charm exists for such an application Juju can declare such requirements both at deploy time and during the lifetime of the application.

> The level of sophistication is limited by the charm; a charm may support multiple storage options (e.g. persistent storage, additional cache). All this allows the user to allocate resources at a granular level. This page will refer to the PostgreSQL charm and the Ceph OSD charm.

It's interesting that I appended the first sentence before reading your review note. Does it help?

This note under 'Storage constraints' also helps nail it:

> 'label' is a string taken from the charm itself. It encapsulates a specific storage option/feature.

It's difficult to explain in detail because of the charm-specific nature of what actually happens. In this specific case I could have just added a second OSD journal and things would work, yet from my Ceph days in the Support group this was never done. There was always a single journal. So this example perhaps has a little extra realism to it. Hopefully that enriches the reader experience and does not inadvertently detract from it.

Regardless, when a journal is added Ceph does indeed do a fair amount of processing before it is ready for usage (as it does when an OSD is added) but I didn't mention that and I don't think I should.